### ISSUE TITLE
Scaling πριν το split

### ISSUE THREAD

**Issue by kostasang** [slp-ntua/patrec-labs#47]
Στην έτοιμη συνάρτηση σας στο αρχείο parser.py πραγματοποιείται fit του standarScaler πάνω σε όλα τα δεδομένα και ύστερα ακολουθεί χωρισμός σε train & test. Στη συνέχεια χωρίζουμε τα δεδομένα train σε train & validation. 

Κανονικά το fit του scaler πρέπει να γίνει μόνο στα τελικά δεδομένα train και όχι στο σύνολο. Θέλετε να συνεχίσουμε ως έχει η να κάνουμε τις κατάλληλες τροποποιήσεις του parser.py ώστε το scaling να γίνεται σωστά;

**Comment by efthymisgeo** [comment]
Nαι έχετε δίκιο ευχαριστούμε για την υπόδειξη. Ανανεώσαμε το `parser.py`

**Comment by vassilispan** [comment]
Νομίζω ότι ακόμα υπάρχει λάθος στον κώδικα. Εμένα μου δείχνει ότι απλά άλλαξαν 4 γραμμές στις οποίες γίνονται οι δηλώσεις και όχι το πραγματικό split.

**Comment by kostasang** [comment]
Πράγματι, το scaling εξακολουθεί να γίνεται πριν το split. 

Επίσης, ο scaler ειναι επιθυμητό να κανει fit στο validation set, που σε αυτη την περίπτωση αποτελεί ποσοστό του train set? Δεν θα ήταν ορθότερο ενα fit transform στο τελικο train, και ύστερα απλα transform σε validation και set? Σε αυτη την πειρπτωση το scaling πρέπει να γίνει εξω απο την συνάρτηση parser.

**Comment by georgepar** [comment]
Good catch παιδιά. Η καλή πρακτική είναι να τα υπολογίζεις όλα στο τελικό train. Αυτό που γινόταν είναι ένας πολύ εύκολος τρόπος να κλέψεις σε speaker independent tasks (γιατί;).
Έκανα ένα refactor τον Parser και τώρα γίνεται το scale εξωτερικά (δείτε την main).

Τώρα ο το συγκεκριμένο dataset δεν έχει πολύ σημασία που θα υπολογίσεις τα statistics (γιατί;).
Ενδεικτικά βάζω τα Mean και τα variances, αν υπολογίσουμε τα statistics στο:

- train + dev + test

```
If using all data to calculate normalization statistics
Normalization will be performed using mean: [-517.71365118   62.17300254   19.00181187    9.64443992  -19.21148097
  -10.88157055]
Normalization will be performed using std: [152.11675793  52.36822512  36.82452731  29.71210532  24.82190459
  23.43786638]
```

- train + dev

```
If using X_train + X_dev to calculate normalization statistics
Normalization will be performed using mean: [-517.66797396   62.20688519   19.05333909    9.66996149  -19.1710291
  -10.74611001]
Normalization will be performed using std: [152.29737032  52.37188378  36.83412308  29.72445749  24.86198985
  23.34840273]
```

- train

```
If using X_train to calculate normalization statistics
Normalization will be performed using mean: [-517.82799022   62.50781171   18.98448756    9.84856086  -19.50537342
  -10.81095257]
Normalization will be performed using std: [152.75770921  52.3203649   36.86289231  29.59151778  24.95244165
  23.54931585]
```

Τώρα για να δείτε που έχει σημασία, μπορείτε να χωρίσετε το dataset με βάση τους speakers:

```
# set(speakers) -> {"nicolas", "george", "theo", "yweweler", "lucas", "jackson"}
X_all = {spk: [] for spk in set(speakers)}
for i, spk in enumerate(speakers):
    X_all[spk].append(frames[i[)
```

και να δοκιμάσετε να βάλετε 4 speakers τυχαία στο train, 1 στο dev και 1 στο test.

**Comment by foivospar** [comment]
Μεταξύ των γραμμών 81 και 82 ξεφυγε ένα `rerurn scaled` ?
